src/lib.rs

```diff
@@ -52,10 +35,10 @@ pub async fn init_eth_adapter(http_url: &str) -> (Provider<Http>, Contract) {
 fn parse_calldata(commit_blocks_fn: &Function, calldata: &[u8]) -> Result<Vec<CommitBlockInfoV1>> {
     let mut parsed_input = commit_blocks_fn
         .decode_input(&calldata[4..])
-        .map_err(|e| ParseError::InvalidCalldata(e.to_string()))?;
+        .map_err(|e| state::ParseError::InvalidCalldata(e.to_string()))?;
```

Comment:
These shared error types here are ugly as hell and don't fell right, but let's fix these separately, if that's ok?